Modules/_sqlite/connection.c

```diff
@@ -1998,8 +2003,10 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self)
         }
         return NULL;
     }
-
-    PyObject *retval = PyObject_CallOneArg(iterdump, (PyObject *)self);
+    PyObject *args[2] = {(PyObject *)self, filter};
+    PyObject *kwnames = PyTuple_New(1);
+    PyTuple_SET_ITEM(kwnames, 0, PyUnicode_FromString("filter"));
+    PyObject *retval = PyObject_Vectorcall(iterdump, args, 1, kwnames);
```

Comment:
You need to add error handling for the `PyTuple_New` and `PyUnicode_FromString` calls.